about/architecture.md

```diff
@@ -7,7 +7,65 @@ permalink: /about/architecture.html
 ---
 
 ### WPE
-WPE is the official WebKit port for embedded.  In WebKit terms a port is...
 
-WPE is uniquely designed for embedded systems in that...
-something something backends
+The [WebKit](https://webkit.org) project is designed to provide programming
+interfaces to Web Browser developers. Depending on the platform and operating
+system the Browser application should run on, the developers should choose the
+right WebKit port suiting their needs. For instance, macOS application
+developers should rely on the WebKit mac ports. Linux desktop developers should
+build their application upon the [GTK port](https://webkitgtk.org).
+
+WPE is the official WebKit port for embedded platforms. WPE is uniquely designed
+for embedded systems in that it doesn't depend on any user-interface toolkit
+such as the traditional Cocoa, GTK, etc toolkits.
+
+<div class="masthead-img" align="center">
+      <img style="width: 400px; max-width: 100%;" src="{{ '/assets/wpe-architecture-diagram.png' | url }}"
+		srcset="{{ '/assets/wpe-architecture-diagram@2x.png' | url }} 2x"
+		alt="Block diagram of the WPE architeture">
+</div>
+
+#### Web page rendering
+
+WPE is considered as an hybrid port, it defers the final web page delivery for
+display to a rendering backend. A traditional port would provide a widget for a
+given toolkit, but WPE opted for a different and more flexible approach.
+
+The common interface between WPEWebKit and its rendering backends is provided by
+[libwpe](https://github.com/WebPlatformForEmbedded/libwpe). On one side, once
+WPEWebKit has a graphical representation of the final composited Web page ready
+for rendering, it invokes a callback function on `libwpe`. On the other side,
```

Comment:
Interesting, I never realized this - does this mean that if you opened the HTML5 living standard single page edition on a slow connection, nothing at all might render for actual minutes?

Reply:
Well, until the engine has an internal representation of the DOM there's not much you can do, the download of the main page shouldn't be too long unless the page contains every resource (images encoded in base64, embedded CSS/JS)...
If the index page is light enough, the engine should be able to render at least a white rectangle while the resources are being downloaded and until enough CSS/JS has been loaded in order to render something useful.